src/data/DataSet.py

```diff
@@ -0,0 +1,53 @@
+from torch.utils.data import Dataset
+from pathlib import Path
+import pandas as pd
+from PIL import Image
+from typing import Union
+import torchvision.transforms as transforms
+
+class SportDataset(Dataset):
+    def __init__(self, csv_file: str | Path, data_dir: str | Path, transformations=None):
+        super().__init__()
+
+        if not isinstance(csv_file, (str, Path)):
+            raise TypeError("csv_file must be of type str or Path")
```

Comment:
Do we want to log errors with logging.error(...)? I think it might be good practice.